rs/orchestrator/src/catch_up_package_provider.rs

```diff
@@ -209,14 +217,79 @@ impl CatchUpPackageProvider {
     // caller.
     async fn fetch_catch_up_package(
         &self,
-        url: Url,
+        node_id: &NodeId,
+        url: String,
         param: Option<CatchUpPackageParam>,
     ) -> Option<pb::CatchUpPackage> {
-        Agent::new_with_client(self.client.clone(), url, Sender::Anonymous)
-            .query_cup_endpoint(param)
+        let body = Bytes::from(
+            param
+                .and_then(|param| serde_cbor::to_vec(&param).ok())
+                .unwrap_or_default(),
+        );
+
+        let client_config = self
+            .crypto_tls_config
+            .client_config(*node_id, self.registry.get_latest_version())
+            .map_err(|e| warn!(self.logger, "Failed to create tls client config: {:?}", e))
+            .ok()?;
+
+        let https = HttpsConnectorBuilder::new()
+            .with_tls_config(client_config)
+            .https_only()
+            .enable_all_versions()
+            .build();
+
+        let client = Client::builder(TokioExecutor::new())
+            .http2_only(true)
```

Comment:
Why only http2? For compatability I would allow both.

Reply:
The `Agent` in the previous version of the cup provider was also using [http2](https://sourcegraph.com/github.com/dfinity/ic@81d5f1809958dd40b573cbc5db3d18b385523f27/-/blob/rs/canister_client/src/http_client.rs?L43) only. Should we allow both considering this?

Reply:
I would allow both